172 unnecessary optional memo fields

Zagan202 opened the pull request:
# Description

- Extends `"has the correct object schema for an existing deposit transaction"` test to check for withdraw fields
- Extends `"has the correct object schema for an existing withdraw transaction"` test to check for deposit fields

Addresses: #172 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [X] `npm run start:dev` Run SEP-24 on testanchor.stellar.org
  - [X] Run SEP-24
  - [ ] Run optional tests
  - [X] DOMAIN testanchor.stellar.org
  - [ ] Run on mainnet
- [X] `DOMAIN=https://testanchor.stellar.org npx jest --roots=cases-SEP24 -i transaction`
- [X] vscode debugger

Zagan202 committed:
Check if exclusive withdraw properties are not present in deposit json

Signed-off-by: Lawrence <lawrence@stellar.org>

Zagan202 committed:
Check if exclusive deposit properties are not present in withdraw transaction response

Signed-off-by: Lawrence <lawrence@stellar.org>

Zagan202 committed:
Replace inccorrect arrayContaining call

- It seems like toContain is sufficient (js is hard)

Signed-off-by: Lawrence <lawrence@stellar.org>

JakeUrban reviewed (approved):
This works but I think modifying the transaction schema object and letting that define the expectation is better.

Zagan202 committed:
Add additionalProperties field to transactions

- This will ensure that theres nothing less or more thats expected
  - examples include deposit properties in withdraw transactions

Signed-off-by: Lawrence <lawrence@stellar.org>

Zagan202 committed:
redo the withdraw or deposit tranaction schema check

- This will create a new json that is "tainted" and check if it will fail
the check

Signed-off-by: Lawrence <lawrence@stellar.org>

JakeUrban commented on cases-SEP24/transaction.test.js:
```diff
@@ -82,6 +82,12 @@ describe("Transaction", () => {
       jwt: jwt,
     });
     await checkTransactionResponse({ json: json, isDeposit: true });
+    const notExpectedJson = json;
```
I think we can remove this file's changes right? The first `checkTransactionResponse()` call in each test will raise an error if the conditions tested below would fail.

Zagan202 replied:
Ah I see your point, yeah I'll revert `cases-SEP24/transaction.test.js`'s changes.

JakeUrban reviewed (approved):
Looks good other than my one comment!

Zagan202 committed:
revert the extensions to the test

- since we altered the expected schema to include an
additional properties flag = false. these tests will fail,
as expected, if there are extra props in the responses.

Signed-off-by: Lawrence <lawrence@stellar.org>
